dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/kcp/DatadogIrExtensionTest.kt

```diff
@@ -0,0 +1,184 @@
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import org.jetbrains.kotlin.ir.declarations.IrClass
+import org.jetbrains.kotlin.ir.declarations.IrModuleFragment
+import org.jetbrains.kotlin.ir.symbols.IrClassSymbol
+import org.jetbrains.kotlin.ir.symbols.IrSimpleFunctionSymbol
+import org.jetbrains.kotlin.name.CallableId
+import org.jetbrains.kotlin.name.ClassId
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@OptIn(FirIncompatiblePluginAPI::class)
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogIrExtensionTest {
+
+    @Mock
+    private lateinit var mockMessageCollector: MessageCollector
+
+    @Mock
+    private lateinit var moduleFragment: IrModuleFragment
+
+    @Mock
+    private lateinit var mockPluginContext: IrPluginContext
+
+    @Mock
+    private lateinit var mockIrSimpleFunctionSymbol: IrSimpleFunctionSymbol
+
+    @Mock
+    private lateinit var mockClassSymbol: IrClassSymbol
+
+    @BeforeEach
+    fun `set up`() {
+        // Deep mock in PluginContext to make `initReference` pass for each transformer
+        val mockOwner = mock<IrClass>()
+        val mockCompanionObject = mock<IrClass>()
+        val mockCompanionSymbol = mock<IrClassSymbol>()
+        whenever(mockPluginContext.referenceFunctions(any<CallableId>())) doReturn listOf(mockIrSimpleFunctionSymbol)
+        whenever(mockPluginContext.referenceClass(any<ClassId>())) doReturn mockClassSymbol
+        whenever(mockClassSymbol.owner) doReturn mockOwner
+        whenever(mockOwner.declarations) doReturn mutableListOf(mockCompanionObject)
+        whenever(mockCompanionObject.isCompanion) doReturn true
+        whenever(mockCompanionObject.symbol) doReturn mockCompanionSymbol
+    }
+
+    @Test
+    fun `M register tag transformer W recordImages option is AUTO`(
+        @Forgery fakeConfiguration: InternalCompilerConfiguration
+    ) {
+        // Given
+        val datadogIrExtension = DatadogIrExtension(
+            mockMessageCollector,
+            fakeConfiguration.copy(
+                recordImages = InstrumentationMode.AUTO
+            )
+        )
+
+        // When
+        datadogIrExtension.generate(moduleFragment, mockPluginContext)
+
+        // Then
+        verify(moduleFragment).accept(any<ComposeTagTransformer>(), eq(null))
```

Comment:
minor: alternatively `isNull` can be used

```suggestion
        verify(moduleFragment).accept(any<ComposeTagTransformer>(), isNull())
```

Reply:
`isNull` throws this compiler warning:
> Returning type parameter has been inferred to Nothing implicitly. Please specify type arguments explicitly to hide this warning. Nothing can produce an exception at runtime.